Mathlib/Analysis/Normed/Unbundled/SpectralNorm.lean

```diff
@@ -841,39 +880,111 @@ def nontriviallyNormedField [CompleteSpace K] : NontriviallyNormedField L where
+  apply congr_arg
+  ext r
+  rw [Multiset.count_replicate]
+  split_ifs with hr
+  · have h : ∀ s : ℝ, s ∈ Multiset.map (spectralMulAlgNorm K E)
+        ((mapAlg K E) (minpoly K x)).roots → r = s := by
+      intro s hs
+      simp only [Multiset.mem_map, mem_roots', ne_eq, IsRoot.def] at hs
+      obtain ⟨a, ha_root, has⟩ := hs
+      rw [← hr, ← has]
+      simp only [spectralMulAlgNorm_def, spectralNorm]
+      rw [← minpoly.eq_of_root (Algebra.IsAlgebraic.isAlgebraic ((algebraMap L E) x))]
+      rw [← ha_root.2, mapAlg_eq_map, minpoly.algebraMap_eq (algebraMap L E).injective, aeval_def,
+        eval_map]
+    rw [Multiset.count_eq_card.mpr h, Multiset.card_map]
+    exact h_deg
+  · rw [Multiset.count_eq_zero_of_notMem]
+    intro hr_mem
+    simp only [Multiset.mem_map, mem_roots', ne_eq, IsRoot.def] at hr_mem
+    obtain ⟨e, he_root, her⟩ := hr_mem
+    have heq : (spectralMulAlgNorm K E) e =
+      (spectralMulAlgNorm K E) ((algebraMap L E) x) := by
+      change spectralNorm K E e = spectralNorm K E (algebraMap L E x)
+      simp only [spectralNorm]
+      rw [minpoly.eq_of_root (Algebra.IsAlgebraic.isAlgebraic ((algebraMap L E) x))]
+      rw [← he_root.2, mapAlg_eq_map, minpoly.algebraMap_eq (algebraMap L E).injective, aeval_def,
+        eval_map]
+    rw [heq] at her
+    exact hr her
+
+/-- For `x : L` with minimal polynomial `f(X) := X^n + a_{n-1}X^{n-1} + ... + a_0` over `K`,
+  the spectral norm of `x` is equal to `‖ a_0 ‖^(1/(degree(f(X))))`. -/
+theorem spectralNorm_eq_root_zero_coeff (x : L) :
+    spectralNorm K L x = ‖(minpoly K x).coeff 0‖ ^ (1 / (minpoly K x).natDegree : ℝ) := by
+  by_cases hx0 : x = 0
+  · simp only [hx0, minpoly.zero, coeff_X_zero, norm_zero, natDegree_X, div_self, ne_eq,
+      one_ne_zero, not_false_iff, spectralNorm_zero, Nat.cast_one, div_self, Real.rpow_one]
+  · set E := (mapAlg K L (minpoly K x)).SplittingField
+    have hspl : Splits (RingHom.id E) (mapAlg K E (minpoly K x)) :=
+      IsSplittingField.IsScalarTower.splits (K := L) E (minpoly K x)
+    have : Algebra.IsAlgebraic L E :=
+      IsSplittingField.IsScalarTower.isAlgebraic E (mapAlg K L (minpoly K x))
+    have : Algebra.IsAlgebraic K E := Algebra.IsAlgebraic.trans K L E
+    rw [one_div, Real.eq_rpow_inv (spectralNorm_nonneg x) (norm_nonneg ((minpoly K x).coeff 0)),
+      Real.rpow_natCast, @spectralNorm.eq_of_tower K _ E,
+      ← @spectralNorm_extends K _ L _ _ ((minpoly K x).coeff 0),
+      @spectralNorm.eq_of_tower K _ E _ _ L, ← spectralMulAlgNorm_def,
+      ← spectralMulAlgNorm_def, Polynomial.coeff_zero_of_isScalarTower,
+      Polynomial.prod_roots_eq_coeff_zero_of_monic_of_splits _ hspl, map_mul, map_pow,
+      map_neg_eq_map, map_one, one_pow, one_mul, spectralNorm_pow_natDegree_eq_prod_roots _ _ x]
+    · simp only [mapAlg_eq_map, Monic, leadingCoeff, coeff_map, natDegree_map]
+      simp [coeff_natDegree, minpoly.monic (Algebra.IsAlgebraic.isAlgebraic x).isIntegral, map_one]
```

Comment:
This should be a one-liner. Can you perhaps add an `iff` version of `Polynomial.monic_of_monic_mapAlg` ?